### ISSUE TITLE
[Arm64] CoreFX tests trigger OutOfMemoryException(s)

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#17173]
This is the first of reportedly several.  I'll try to keep them all in this issue.

```
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Composition.TypedParts.Tests
Discovered:  System.Composition.TypedParts.Tests
Starting:    System.Composition.TypedParts.Tests
   System.Composition.TypedParts.Tests.ReflectionTests.MultiThreadedGetExportsWorkWithImportingConstuctor [FAIL]
      System.AggregateException : One or more errors occurred. (Insufficient memory within specified address space range to continue the execution of the program.)
      ---- System.OutOfMemoryException : Insufficient memory within specified address space range to continue the execution of the program.
      Stack Trace:
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/tests/ReflectionTests.cs(58,0): at System.Composition.TypedParts.Tests.ReflectionTests.MultiThreadedGetExpoOutOfMemoryExceptionrtsWorkWithImportingConstuctor()
         ----- Inner Stack Trace -----
            at System.Runtime.CompilerServices.RuntimeHelpers._CompileMethod(IRuntimeMethodInfo method)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs(374,0): at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType, Object target)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs(185,0): at System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs(206,0): at System.Linq.Expressions.Expression`1.Compile(Boolean preferInterpretation)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredPart.cs(167,0): at System.Composition.TypedParts.Discovery.DiscoveredPart.GetActivator(DependencyAccessor definitionAccessor, IEnumerable`1 dependencies)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredExport.cs(42,0): at System.Composition.TypedParts.Discovery.DiscoveredExport.<>c__DisplayClass11_0.<GetExportDescriptorPromise>b__1(IEnumerable`1 deps)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/shared/System/Lazy.cs(324,0): at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/shared/OutOfMemoryExceptionSystem/Lazy.cs(408,0): at System.Lazy`1.CreateValue()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorPromise.cs(86,0): at System.Composition.Hosting.Core.ExportDescriptorPromise.GetDescriptor()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredPart.cs(155,0): at System.Composition.TypedParts.Discovery.DiscoveredPart.GetActivator(DependencyAccessor definitionAccessor, IEnumerable`1 dependencies)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredExport.cs(42,0): at System.Composition.TypedParts.Discovery.DiscoveredExport.<>c__DisplayClass11_0.<GetExportDescriptorPromise>b__1(IEnumerable`1 deps)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/shared/System/Lazy.cs(324,0): at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/shared/System/Lazy.cs(408,0): at System.Lazy`1.CreateValue()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorPromise.cs(86,0): at System.Composition.Hosting.Core.ExportDescriptorPromise.GetDescriptor()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq/src/System/Linq/Select.cs(240,0): at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorRegistryUpdate.cs(50,0): at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.Execute(CompositionContract contract)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorRegistry.cs(32,0): at System.Composition.Hosting.Core.ExportDescriptorRegistry.TryGetSingleForExport(CompositionContract exportKey, ExportDescriptor& defaultForExport)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Hosting/src/System/Composition/Hosting/Core/LifetimeContext.cs(196,0): at System.Composition.Hosting.Core.LifetimeContext.TryGetExport(CompositionContract contract, Object& export)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Runtime/src/System/Composition/CompositionContext.cs(150,0): at System.Composition.CompositionContext.GetExport(CompositionContract contract)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.Runtime/src/System/Composition/CompositionContext.cs(50,0): at System.Composition.CompositionContext.GetExport[TExport](String contractName)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/tests/ReflectionTests.cs(78,0): at System.Composition.TypedParts.Tests.ReflectionTests.GetExport()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Composition.TypedParts/tests/ReflectionTests.cs(67,0): at System.Composition.TypedParts.Tests.ReflectionTests.Run(ManualResetEventSlim evnt, ConcurrentBag`1 errors)
Finished:    System.Composition.TypedParts.Tests
```

**Comment by sdmaclea** [comment]
```
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Dynamic.Runtime.Tests
Discovered:  System.Dynamic.Runtime.Tests
Starting:    System.Dynamic.Runtime.Tests
   ManagedTests.DynamicCSharp.Conformance.dynamic.Variance.assign.assignment07.assignment07.C.DynamicCSharpRunTest [FAIL]
      System.OutOfMemoryException : Insufficient memory within specified address space range to continue the execution of the program.
      Stack Trace:
            at System.Runtime.CompilerServices.RuntimeHelpers._CompileMethod(IRuntimeMethodInfo method)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs(374,0): at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType, Object target)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs(185,0): at System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs(206,0): at System.Linq.Expressions.Expression`1.Compile(Boolean preferInterpretation)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Runtime/CompilerServices/CallSiteBinder.cs(144,0): at System.Runtime.CompilerServices.CallSiteBinder.BindCore[T](CallSite`1 site, Object[] args)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Dynamic/UpdateDelegates.Generated.cs(408,0): at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Dynamic.Runtime/tests/Dynamic.Variance/Conformance.dynamic.Variance.assign.cs(529,0): at ManagedTests.DynamicCSharp.Conformance.dynamic.Variance.assign.assignment07.assignment07.C.MainMethod()
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Dynamic.Runtime/tests/Dynamic.Variance/Conformance.dynamic.Variance.assign.cs(505,0): at ManagedTests.DynamicCSharp.Conformance.dynamic.Variance.assign.assignment07.assignment07.C.DynamicCSharpRunTest()
```

**Comment by sdmaclea** [comment]
```
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Linq.Expressions.Tests
Discovered:  System.Linq.Expressions.Tests
Starting:    System.Linq.Expressions.Tests
   System.Linq.Expressions.Tests.LiftedModuloNullableTests.CheckLiftedModuloNullableSByteTest(useInterpreter: False) [FAIL]
      System.OutOfMemoryException : Insufficient memory within specified address space range to continue the execution of the program.
      Stack Trace:
            at System.Runtime.CompilerServices.RuntimeHelpers._CompileMethod(IRuntimeMethodInfo method)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs(374,0): at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType, Object target)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs(185,0): at System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs(206,0): at System.Linq.Expressions.Expression`1.Compile(Boolean preferInterpretation)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/tests/Lifted/LiftedModuloNullableTests.cs(373,0): at System.Linq.Expressions.Tests.LiftedModuloNullableTests.VerifyModuloNullableSByte(Nullable`1 a, Nullable`1 b, Boolean useInterpreter)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/tests/Lifted/LiftedModuloNullableTests.cs(118,0): at System.Linq.Expressions.Tests.LiftedModuloNullableTests.CheckLiftedModuloNullableSByteTest(Boolean useInterpreter)
   System.Linq.Expressions.Tests.LambdaTests.InvokeComputedLambda [SKIP]
      870811
   System.Linq.Expressions.Tests.Compiler_Tests.ArrayInitializedWithCapturedInstance [SKIP]
      870811
   System.Linq.Expressions.Tests.Compiler_Tests.CallOnCapturedInstance [SKIP]
      870811
   System.Linq.Expressions.Tests.Compiler_Tests.NewExpressionwithMemberAssignInit [SKIP]
      870811
   System.Linq.Expressions.Tests.Compiler_Tests.TestAndAlso [SKIP]
      870811
   System.Linq.Expressions.Tests.LambdaModuloTests.LambdaModuloIntTest(useInterpreter: True) [FAIL]
      System.OutOfMemoryException : Insufficient memory within specified address space range to continue the execution of the program.
      Stack Trace:
            at Thunk(Func`2 )
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/tests/Lambda/LambdaModuloTests.cs(618,0): at System.Linq.Expressions.Tests.LambdaModuloTests.VerifyModuloInt(Int32 a, Int32 b, Boolean useInterpreter)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/tests/Lambda/LambdaModuloTests.cs(61,0): at System.Linq.Expressions.Tests.LambdaModuloTests.LambdaModuloIntTest(Boolean useInterpreter)
Finished:    System.Linq.Expressions.Tests

=== TEST EXECUTION SUMMARY ===
   System.Linq.Expressions.Tests  Total: 35203, Errors: 0, Failed: 2, Skipped: 5, Time: 217.945s
```

**Comment by sdmaclea** [comment]
```
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Linq.Queryable.Tests
Discovered:  System.Linq.Queryable.Tests
Starting:    System.Linq.Queryable.Tests
   System.Linq.Tests.TakeLastTests.TakeLast(equivalent: [0, 1, 2], count: -55) [FAIL]
      System.OutOfMemoryException : Insufficient memory within specified address space range to continue the execution of the program.
      Stack Trace:
            at System.Runtime.CompilerServices.RuntimeHelpers._CompileMethod(IRuntimeMethodInfo method)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs(374,0): at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType, Object target)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs(185,0): at System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs(206,0): at System.Linq.Expressions.Expression`1.Compile(Boolean preferInterpretation)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Queryable/src/System/Linq/EnumerableQuery.cs(114,0): at System.Linq.EnumerableQuery`1.GetEnumerator()
   System.Linq.Tests.SkipLastTests.SkipLast(source: [0], count: 89) [FAIL]
      System.OutOfMemoryException : Insufficient memory within specified address space range to continue the execution of the program.
      Stack Trace:
            at System.Runtime.CompilerServices.RuntimeHelpers._CompileMethod(IRuntimeMethodInfo method)
         /home/vmjenkins/workspace/Dotnet/build_and_test/src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs(374,0): at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType, Object target)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs(185,0): at System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs(206,0): at System.Linq.Expressions.Expression`1.Compile(Boolean preferInterpretation)
         /home/vmjenkins/workspace/Dotnet/build_corefx/src/System.Linq.Queryable/src/System/Linq/EnumerableQuery.cs(114,0): at System.Linq.EnumerableQuery`1.GetEnumerator()
Finished:    System.Linq.Queryable.Tests

=== TEST EXECUTION SUMMARY ===
   System.Linq.Queryable.Tests  Total: 1324, Errors: 0, Failed: 2, Skipped: 0, Time: 5.609s
```

**Comment by jkotas** [comment]
The jump stub space reservation scheme need to be enabled for ARM64. It is under AMD64 ifdefs right now. You can get an idea about the affected code from: https://github.com/dotnet/coreclr/pull/15296

Perhaps we need a new `FEATURE_JUMPSTUB`.

**Comment by sdmaclea** [comment]
You can assign this to me.  I'd like this fixed for 2.1
